tests/conftest.py

```diff
@@ -430,3 +471,178 @@ def plates_lookup_with_samples(samples, priority_samples):
+        status=cherrytrack_mock_run_info_status,
+    )
+    yield
+
+
+@pytest.fixture
+def baracoda_mock_barcodes_group(app, mocked_responses, baracoda_mock_responses, baracoda_mock_status):
+    for centre_prefix in baracoda_mock_responses.keys():
+        if baracoda_mock_responses[centre_prefix] is not None:
+            num_samples = len(baracoda_mock_responses[centre_prefix]["barcodes_group"]["barcodes"])
+            baracoda_url = (
+                f"http://{app.config['BARACODA_URL']}" f"/barcodes_group/{centre_prefix}/new?count={num_samples}"
+            )
+            mocked_responses.add(
+                responses.POST,
+                baracoda_url,
+                json=baracoda_mock_responses[centre_prefix],
+                status=baracoda_mock_status,
+            )
+    yield
+
+
+@pytest.fixture
+def cherrytrack_mock_source_plates_status():
+    return HTTPStatus.OK
+
+
+@pytest.fixture
+def cherrytrack_mock_run_info_status():
+    return HTTPStatus.OK
+
+
+@pytest.fixture
+def cherrytrack_mock_destination_plate_status():
+    return HTTPStatus.OK
+
+
+@pytest.fixture
+def baracoda_mock_status():
+    return HTTPStatus.CREATED
+
+
+@pytest.fixture
+def cherrytrack_mock_source_plates(
+    app,
+    mocked_responses,
+    source_barcode,
+    destination_barcode,
+    cherrytrack_source_plates_response,
+    cherrytrack_mock_source_plates_status,
+):
+    source_plates_url = f"{app.config['CHERRYTRACK_URL']}/source-plates/{source_barcode}"
+    mocked_responses.add(
+        responses.GET,
+        source_plates_url,
+        json=cherrytrack_source_plates_response,
+        status=cherrytrack_mock_source_plates_status,
+    )
+    yield
+
+
+@pytest.fixture
+def cherrytrack_mock_destination_plate(
+    app,
+    mocked_responses,
+    destination_barcode,
+    cherrytrack_destination_plate_response,
+    cherrytrack_mock_destination_plate_status,
+):
+    destination_plate_url = f"{app.config['CHERRYTRACK_URL']}/destination-plates/{destination_barcode}"
+    mocked_responses.add(
+        responses.GET,
+        destination_plate_url,
+        json=cherrytrack_destination_plate_response,
+        status=cherrytrack_mock_destination_plate_status,
+    )
+    yield
+
+
+@pytest.fixture
+def cherrytrack_run_info_response(run_id):
+    return {
+        "data": {
+            "id": run_id,
+            FIELD_CHERRYTRACK_USER_ID: "user1",
+            FIELD_CHERRYTRACK_LIQUID_HANDLER_SERIAL_NUMBER: "aLiquidHandlerSerialNumber",
+            FIELD_CHERRYTRACK_AUTOMATION_SYSTEM_MANUFACTURER: "biosero",
+            FIELD_CHERRYTRACK_AUTOMATION_SYSTEM_NAME: "CPA",
+        }
+    }
+
+
+@pytest.fixture
+def cherrytrack_destination_plate_response(destination_barcode, source_barcode, run_id):
+    return build_cherrytrack_destination_plate_response(destination_barcode, source_barcode, run_id)
+
+
+def cherrytrack_destination_plate_response_duplicated_wells(cherrytrack_destination_plate_response):
```

Comment:
Functions in this file which are not fixtures does not feel right. Also, the dictionary is being changed in place so it does not need to be returned.